3-scalability/README.md

```diff
@@ -0,0 +1,427 @@
+
+default_tags = {
+  Environment = "production"
+  Project     = "mass-ingest"
+}
+```
+
+### 3. Store secrets in AWS Secrets Manager
+
+```bash
+aws secretsmanager create-secret \
+  --name mass-ingest/moderne-token \
+  --secret-string "your-moderne-token"
+
+aws secretsmanager create-secret \
+  --name mass-ingest/publish-user \
+  --secret-string "your-artifactory-user"
+
+aws secretsmanager create-secret \
+  --name mass-ingest/publish-password \
+  --secret-string "your-artifactory-password"
+```
+
+### 4. Deploy infrastructure
+
+```bash
+cd terraform
+terraform init
+terraform plan
+terraform apply
+```
+
+This creates:
+- Batch compute environment (EC2 instances)
+- Batch job queue
+- Job definitions (chunk + processor)
+- IAM roles and policies
+- EventBridge schedule (daily at midnight UTC)
+- Security groups
+- CloudWatch log groups
+
+### 5. Prepare repos.csv
+
+Create or edit `../repos.csv` with your repositories:
+
+```csv
+cloneUrl,branch,origin,path
+https://github.com/org/repo1,main,github.com,org/repo1
+https://github.com/org/repo2,main,github.com,org/repo2
+```
+
+Required columns:
+- `cloneUrl` - Full HTTPS clone URL
+- `branch` - Branch to build
+- `origin` - Source control host (e.g., github.com)
+- `path` - Repository path (e.g., org/repo)
+
+### 6. Configure repository authentication (if needed)
+
+For private repositories, you have several options for providing git credentials to AWS Batch workers.
+
+**Option A: AWS Secrets Manager (recommended)**
+
+Store credentials in Secrets Manager and inject them into the container:
+
+```bash
+# Create secret with git credentials
+aws secretsmanager create-secret \
+  --name mass-ingest/git-credentials \
+  --secret-string "https://username:token@github.com
+https://username:token@gitlab.com"
+
+# Add to terraform.tfvars
+git_credentials_secret_arn = "arn:aws:secretsmanager:region:account:secret:mass-ingest/git-credentials"
+```
+
+Then modify the job definition to mount the secret as a file or environment variable.
+
+**Option B: EFS mount (recommended for SSH keys)**
+
+1. Create an EFS filesystem
+2. Mount it to an EC2 instance and copy `.git-credentials` or `.ssh` directory
+3. Configure Batch job definition to mount the EFS volume at runtime
+4. Set proper mount points: `/root/.git-credentials` or `/root/.ssh`
+
+**Option C: Bake into image (use with caution)**
+
+If all workers need identical credentials and your image registry has strict access controls:
+
+```bash
+# Build with credentials
+docker build -t mass-ingest:latest \
+  --secret id=git-credentials,src=.git-credentials \
+  ..
+
+# Push to secure registry
+docker push <account-id>.dkr.ecr.us-east-1.amazonaws.com/mass-ingest:latest
+```
+
+⚠️ **WARNING**: This bakes secrets into the image. Only use if your ECR repository has proper IAM access controls. Consider using AWS ECR's image scanning to detect exposed credentials.
```

Comment:
I would leave this option out completely. I know we're being very clear that this is a bad idea, so probably best not to even leave it as an option.